aidews.go

```diff
@@ -5,31 +5,44 @@ package aidews
 import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
-	"github.com/aws/aws-sdk-go/aws/session"
 )
 
-// Session returns an aws session.
-// The region and role_arn parameters are optional. If neither are given the
-// session returned is built with a blank config. If region is given, the config
-// used to get the session includes the region. If role_arn given, we first STS,
-// then get a session in that region using the credentials from the STS call.
-//
-// All Sessions are constructed using the SharedConfigEnable setting allowing
-// the use of local credential resolution.
-func Session(region, roleARN *string) *session.Session {
+func session(session *session.Session, region, roleARN *string) *session.Session {
 	cfg := aws.Config{
 		Region: region,
 	}
+	if session == nil {
+		session = sessionWithConfig(cfg)
+	}
 	if roleARN != nil {
 		creds := stscreds.NewCredentials(
-			sessionWithConfig(cfg),
+			session,
 			*roleARN,
 		)
 		cfg.Credentials = creds
 	}
 	return sessionWithConfig(cfg)
 }
 
+// Session returns an aws session.
+// The region and role_arn parameters are optional. If neither are given the
+// session returned is built with a blank config. If region is given, the config
+// used to get the session includes the region. If role_arn given, we first STS,
+// then get a session in that region using the credentials from the STS call.
+//
+// All Sessions are constructed using the SharedConfigEnable setting allowing
+// the use of local credential resolution.
+func Session(region, roleARN *string) *session.Session {
+	return session(nil, region, roleARN)
+}
+
+// SessionHop returns an aws session constructed from a given Session.
+// This is very similar to Session, but allows hopping from a given session, to
```

Comment:
It might be good to use the assumerole terminology here for the sake of clarity

Reply:
Very good, sir. It shall be done.

Reply:
Updated the documentation, and added the following example text.

``` go
start := Session(region, startingRoleARN)
hop1 := SessionHop(start, region, hop1ARN)
hop2 := SessionHop(hop1, region, hop2ARN)
destination := SessionHop(hop2, region, destARN)
```